web-services-app/src/test/java/edu/unc/lib/boxc/web/services/rest/IiifV3ManifestControllerTest.java

```diff
@@ -0,0 +1,138 @@
+package edu.unc.lib.boxc.web.services.rest;
+
+import edu.unc.lib.boxc.auth.api.Permission;
+import edu.unc.lib.boxc.auth.api.exceptions.AccessRestrictionException;
+import edu.unc.lib.boxc.auth.api.models.AccessGroupSet;
+import edu.unc.lib.boxc.auth.api.services.AccessControlService;
+import edu.unc.lib.boxc.auth.fcrepo.models.AccessGroupSetImpl;
+import edu.unc.lib.boxc.auth.fcrepo.services.GroupsThreadStore;
+import edu.unc.lib.boxc.model.api.ResourceType;
+import edu.unc.lib.boxc.model.api.ids.PID;
+import edu.unc.lib.boxc.model.fcrepo.ids.PIDs;
+import edu.unc.lib.boxc.search.solr.models.ContentObjectSolrRecord;
+import edu.unc.lib.boxc.search.solr.models.DatastreamImpl;
+import edu.unc.lib.boxc.web.common.services.AccessCopiesService;
+import edu.unc.lib.boxc.web.services.processing.IiifV3ManifestService;
+import edu.unc.lib.boxc.web.services.rest.exceptions.RestResponseEntityExceptionHandler;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.openMocks;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * @author bbpennel
+ */
+public class IiifV3ManifestControllerTest {
+    private static final String IIIF_BASE = "http://example.com/iiif/v3/";
+    private static final String SERVICES_BASE = "http://example.com/services/";
+    private static final String ACCESS_BASE = "http://example.com/";
+
+    private static final String OBJECT_ID = "f277bb38-272c-471c-a28a-9887a1328a1f";
+    private static final PID OBJECT_PID = PIDs.get(OBJECT_ID);
+    private final static String USERNAME = "test_user";
+    private final static AccessGroupSet GROUPS = new AccessGroupSetImpl("adminGroup");
+
+    @InjectMocks
+    private IiifV3ManifestController manifestController;
+
+    @Mock
+    private AccessControlService accessControlService;
+
+    @Mock
+    private AccessCopiesService accessCopiesService;
+
+    private IiifV3ManifestService manifestService;
+
+    private MockMvc mockMvc;
+
+    private AutoCloseable closeable;
+
+    @BeforeEach
+    public void setup() {
+        closeable = openMocks(this);
```

Comment:
do you not have to close this after each because it's autocloseable?

Reply:
Nope, I forgot to add the close statements, good catch